### DISCUSSION TITLE
Welcome to chrome-extension-boilerplate-react Discussions!

### DISCUSSION THREAD

**Discussion by lxieyang** [lxieyang/chrome-extension-boilerplate-react#21] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by heytulsiprasad** [comment]
Hey @lxieyang , thanks for this amazing starter.

However, I'd like to know if there's any issues while uploading this to actual chrome web store? And with overall bundle size?

I kinda am using lot of dependencies for mine, but don't know if that's a good practice and follow chrome standards. If there's any common standards to take care of, do share or link them. Thanks again.

**Comment by istvan73** [comment]
Hi @lxieyang , I just dropped in to say a huuuge THANK YOU!

I've been looking for the exact same features you offered on a plate (on a boilerplate, to be specific :D). 
Also, seeing how much thought you gave this project made me smile.

Thanks a lot again, and if I can do anything besides indicating your name and project in my upcoming extension please let me know.
István

**Comment by widowmaker110** [comment]
this saved me so much time. thank you!!!

**Comment by gunbamz** [comment]
Hi @lxieyang , thanks for this extra-ordinary boilerplate.
Could you please upgrade to React 18?

**Comment by limbuscode** [comment]
Hello folks, 
I'd like to load a custom web page built in react through an action listener on background.js, like this:

chrome.action.onClicked.addListener(function() {
  chrome.tabs.create({url: 'index.html'});
});

How do I do that?
Do I need to configure webpack and all that stuff to make it work?

**Reply by limbuscode** [reply]
oh, and an error occurred on that background.js file when I ran the extension
Uncaught TypeError: Cannot read properties of undefined (reading 'onClicked')

**Reply by limbuscode** [reply]
chrome.action undefined?? Was chrome API not supposed to be accessible on background.js?

**Reply by Aryaman73** [reply]
@limbuscode did you ever figure this out? It seems like Chrome API is inaccessible in `popup.js` as well :/

**Reply by limbuscode** [reply]
You need to add this line at the beginning of background.js to make it work:
`/*global chrome*/`

* Using latest standard npm create react app kind of command to initialize a project
* Place background.js into "public" folder.

**Reply by Aryaman73** [reply]
Thanks 😄

**Comment by akbar-animaker** [comment]
Is there any possibility to share the data across the pages? like redux?

**Reply by Schleidens** [reply]
local storage or context

**Comment by noximus** [comment]
I am trying to see my content scripts run on a webpage locally so i don't have to upload it to the extensions every time i save the file.  The hot-reloading rebuilds the page yeah... but that does nothing If i can't actually see the extension pop up.

**Comment by itz-arnav** [comment]
How to stop modifying new tab?
